railties/lib/rails/commands/rake/rake_command.rb

```diff
@@ -21,11 +21,26 @@ def perform(task, args, config)
             if Rails.respond_to?(:root)
               rake.options.suppress_backtrace_pattern = /\A(?!#{Regexp.quote(Rails.root.to_s)})/
             end
+            unless exit_if_task_does_not_exist
+              # Check if the task does not exist, and safely abort in that case.
+              # If we don't do this, `standard_exception_handling` will exit(false) if the task doesn't exist.
+              return unless task_exists?(rake, task)
```

Comment:
This double unless is hard to parse. Go with a complete conditional:

```ruby
if exit_if_task_does_not_exist && !task_exists?(rake, task)
  return
else
  rake.standard_exception_handling { rake.top_level }
end
```

Reply:
Yeah I agree. I renamed the variable to something easier to parse.